iothub_client/tests/iothubtr_amqp_msgr_ut/iothubtr_amqp_msgr_ut.c

```diff
@@ -959,6 +959,79 @@ static AMQP_MESSENGER_HANDLE create_and_start_messenger2(AMQP_MESSENGER_CONFIG*
     return handle;
 }
 
+char* umock_stringify_AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO(const AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO* value)
+{
+    char* result = (char*)TEST_malloc(1);
+    (void)value;
+    result[0] = '\0';
+    return result;
+}
+
+int umock_are_equal_AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO(const AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO* left, const AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO* right)
+{
+    int result;
+
+    if (((left == NULL) && (right != NULL)) || ((right == NULL) && (left != NULL)))
```

Comment:
You need one more check in case both are NULL, otherwise you will get a SEGFAULT on the next check when derefencing a NULL pointer.

Reply:
True, I missed that check, if both are NULL we are ok since they are equal to each other, but since I'm checking members below that would SEGFUALT